### PR TITLE
cpu_info: mock input in unit tests

### PR DESCRIPTION
**Pull request by petreeftime**
To allow cpu_info unit tests to run even on systems without Nitro
Enclaves, or with Nitro Enclaves in a different configuration, allow
reading CpuInfo from a BufRead, to mock the expected output.

Signed-off-by: Petre Eftime <epetre@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sabin-rapan** [comment]
CI build fails with
```
=================== cargo about ===========================
+ make nitro-about
docker run \
	-v "$(readlink -f .)":/nitro_src \
	-v "$(readlink -f ./build)":/nitro_build \
	"nitro_cli:1.0" bin/bash -c \
		'source /root/.cargo/env && \
		cargo about generate \
		--manifest-path=/nitro_src/Cargo.toml \
		/nitro_src/about.hbs | \
		diff /nitro_src/THIRD_PARTY_LICENSES_RUST_CRATES.html -'
2022-08-31 06:10:47 [[33mWARN[0m] clearlydefined detected license in 'LICENSE' for crate 'half 1.8.2', but it we failed to determine what its license was[0m
2022-08-31 06:11:23 [[33mWARN[0m] failed to request license information from clearly defined: error sending request for url (https://api.clearlydefined.io/definitions): operation timed out[0m
2022-08-31 06:11:36 [[33mWARN[0m] clearlydefined detected license in 'LICENSE' for crate 'typenum 1.15.0', but it we failed to determine what its license was[0m
2022-08-31 06:11:55 [[33mWARN[0m] crate 'driver-bindings 0.1.0' doesn't have a license field[0m
2022-08-31 06:11:55 [[33mWARN[0m] crate 'eif_loader 0.1.0' doesn't have a license field[0m
2022-08-31 06:11:55 [[33mWARN[0m] crate 'enclave_build 0.1.0' doesn't have a license field[0m
2022-08-31 06:11:57 [[33mWARN[0m] unable to synthesize license expression for 'driver-bindings 0.1.0': no `license` specified, and no license files were found[0m
2022-08-31 06:11:57 [[33mWARN[0m] unable to synthesize license expression for 'eif_loader 0.1.0': no `license` specified, and no license files were found[0m
2022-08-31 06:11:57 [[33mWARN[0m] unable to synthesize license expression for 'enclave_build 0.1.0': no `license` specified, and no license files were found[0m
2022-08-31 06:11:57 [[33mWARN[0m] unable to find text for license 'Apache-2.0' for crate 'chrono 0.4.19', falling back to canonical text[0m
2022-08-31 06:11:57 [[33mWARN[0m] unable to find text for license 'Apache-2.0' for crate 'half 1.8.2', falling back to canonical text[0m
2022-08-31 06:11:57 [[33mWARN[0m] unable to find text for license 'Apache-2.0' for crate 'openssl 0.10.38', falling back to canonical text[0m
2022-08-31 06:11:57 [[33mWARN[0m] unable to find text for license 'Apache-2.0' for crate 'tinyvec_macros 0.1.0', falling back to canonical text[0m
48c48
<             <li><a href="#Apache-2.0">Apache License 2.0</a> (27)</li>
---
>             <li><a href="#Apache-2.0">Apache License 2.0</a> (26)</li>
4869a4870
>                     <li><a href=" https://github.com/sfackler/rust-openssl ">openssl 0.10.38</a></li>
4939,4961d4939
< 
< Unless required by applicable law or agreed to in writing, software
< distributed under the License is distributed on an &quot;AS IS&quot; BASIS,
< WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
< See the License for the specific language governing permissions and
< limitations under the License.
< </pre>
<             </li>
<             <li class="license">
<                 <h3 id="Apache-2.0">Apache License 2.0</h3>
<                 <h4>Used by:</h4>
<                 <ul class="license-used-by">
<                     <li><a href=" https://github.com/sfackler/rust-openssl ">openssl 0.10.38</a></li>
<                 </ul>
<                 <pre class="license-text">Copyright 2011-2017 Google Inc.
<           2013 Jack Lloyd
<           2013-2014 Steven Fackler
< 
< Licensed under the Apache License, Version 2.0 (the &quot;License&quot;);
< you may not use this file except in compliance with the License.
< You may obtain a copy of the License at
< 
<     http://www.apache.org/licenses/LICENSE-2.0
Makefile:253: recipe for target 'nitro-about' failed
make: *** [nitro-about] Error 1
```

I know it's unrelated to your RP, but do you mind fixing it as well? :D

**Comment by petreeftime** [comment]
I don't mind, but there is an issue, in the sense that `make update-third-party-licenses-rust-crates-html` doesn't seem to generate any changes for me.

**Review by sabin-rapan — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -105,16 +105,8 @@ driver-deps:
...
-# a file with the same name via the touch command. This
-# change is required in order to capture the timestamp
-# of the rule.
-.build-container: tools/Dockerfile1804.${HOST_MACHINE}
-	docker image build -t $(CONTAINER_TAG) -f tools/Dockerfile1804.${HOST_MACHINE} tools/
-	touch $@
-
-build-container: .build-container
+build-container:
+	@docker image build --pull --quiet -t $(CONTAINER_TAG) -f tools/Dockerfile1804.${HOST_MACHINE} tools/
```
Comment (shtaked) [inline Makefile:109]: won't it be too long to rebuild the container every time?

Comment (petreeftime) [reply Makefile:109]: It shouldn't build every time, it only builds if it's not in the cache, or if Dockerfile has changed or if the base image has changed.

**Review by shtaked — APPROVED** [review]

**Pull request merged**